src/main/cpp/chassis/states/RobotDrive.cpp

```diff
@@ -13,64 +13,42 @@
 // OR OTHER DEALINGS IN THE SOFTWARE.
 //====================================================================================================================================================
 
-#include <string>
-
-// FRC Includes
-#include "frc/filter/SlewRateLimiter.h"
-#include "units/velocity.h"
-#include "units/angle.h"
-
-// Team302 Includes
-#include "chassis/definitions/ChassisConfig.h"
-#include "chassis/definitions/ChassisConfigMgr.h"
-#include "chassis/ChassisMovement.h"
-#include "chassis/modifiers/AntiTip.h"
 #include "chassis/states/RobotDrive.h"
-#include "utils/FMSData.h"
 #include "utils/logging/debug/Logger.h"
 
-using std::string;
-
-RobotDrive::RobotDrive(SwerveChassis *chassis) : ISwerveDriveState::ISwerveDriveState(),
-                                                 m_chassis(chassis),
-                                                 m_flState(),
-                                                 m_frState(),
-                                                 m_blState(),
-                                                 m_brState(),
-                                                 m_wheelbase(units::length::inch_t(20.0)),
-                                                 m_wheeltrack(units::length::inch_t(20.0)),
-                                                 m_maxspeed(units::velocity::feet_per_second_t(1.0))
+// Note the simplified constructor and AddRequirements call
+RobotDrive::RobotDrive(subsystems::CommandSwerveDrivetrain *chassis,
+                       TeleopControl *controller,
+                       units::velocity::meters_per_second_t maxSpeed,
+                       units::angular_velocity::degrees_per_second_t maxAngularRate) : m_chassis(chassis),
+                                                                                       m_controller(controller),
+                                                                                       m_maxSpeed(maxSpeed),
+                                                                                       m_maxAngularRate(maxAngularRate)
 {
-    if (m_chassis != nullptr)
-    {
-        m_wheelbase = m_chassis->GetWheelBase();
-        m_wheeltrack = m_chassis->GetTrack();
-        m_maxspeed = m_chassis->GetMaxSpeed();
-    }
-    else
-    {
-        Logger::GetLogger()->LogData(LOGGER_LEVEL::ERROR_ONCE, string("RobotDrive"), string("Chassis"), string("nullptr"));
-    }
+    AddRequirements(m_chassis);
 }
 
-std::string RobotDrive::GetDriveStateName() const
+void RobotDrive::Execute()
 {
-    return std::string("RobotDrive");
+    double forward = m_controller->GetAxisValue(TeleopControlFunctions::HOLONOMIC_DRIVE_FORWARD);
```

Comment:
Is there now a separate auton drive?  If so, we should probably rename this; otherwise, auton shouldn't care about the controller.

Maybe this one is holonomic drive equivalent

Reply:
No, I was thinking if we are only using this for teleop, it might be clearer to call it TeleopDrive or JoystickDrive because robot drive has some connotations that this really isn't obeying

Reply:
rename to TeleopRobotDrive and TeleopFieldDrive